### PR TITLE
Add mate-power-manager

### PR DESCRIPTION
**Pull request by pyhalov**
Depends on https://github.com/OpenIndiana/oi-userland/pull/3869

### CODE REVIEW

**Comment by pyhalov** [comment]
mate-power manager still can try to suspend PC. It should avoid doing so, as this will result in 'Failed to supsend messages'

**Comment by Mno-hime** [comment]
Starting `mate-power-manager` when one is already running end with an ugly traceback:
```
jack@openindiana:/jack$ mate-power-manager 
TI:09:22:45 TH:4cf860       FI:/export/home/alp/srcs/oi-userland/components/desktop/mate/mate-power-manager/mate-power-manager-1.18.1/src/gpm-main.c        FN:main,246
 - Power Manager is already running in this session.
Traceback:
        /usr/bin/mate-power-manager'egg_error_real+0xd4 [0x429394]
        /usr/bin/mate-power-manager'main+0x3d4 [0x41c7c4]
        /usr/bin/mate-power-manager'_start_crt+0x83 [0x413363]
        /usr/bin/mate-power-manager'_start+0x18 [0x4132c8]
```
Would be nice if avoided.

File: components/desktop/mate/mate-power-manager/patches/03-icon_policy.patch [hunk 1]
```diff
@@ -0,0 +1,11 @@
+--- mate-power-manager-1.18.1/data/org.mate.power-manager.gschema.xml.in.1	2018-01-11 12:58:03.505128814 +0000
++++ mate-power-manager-1.18.1/data/org.mate.power-manager.gschema.xml.in	2018-01-11 12:59:58.776390481 +0000
+@@ -336,7 +336,7 @@
+       <summary>If preferences and statistics items should be shown in the context menu</summary>
+     </key>
+     <key name="icon-policy" enum="org.mate.power-manager.IconPolicy">
+-      <default>'present'</default>
++      <default>'always'</default>
```
Comment (Mno-hime) [inline components/desktop/mate/mate-power-manager/patches/03-icon_policy.patch:8]: I'd actually rather see the icon present only when battery is present - I don't see why would anyone needed it without battery inside. (Though is good to have this setting for testing.)

**Comment by Mno-hime** [comment]
AC adapter is always "online", even when the cable is disconnected and laptop is running on battery (if I read the attribute correctly):
always_online

Don't see this.

**Comment by Mno-hime** [comment]
The "Ask me" option of what to do when the power button is pressed is not effective - I got shutdown as usual:
always_display_icon

**Comment by Mno-hime** [comment]
There's a discrepancy in what DDU and MATE report on the battery capacity. It seem to me that DDU is right in this regard (see the selected items in both windows):
mate_v_ddu_watts

**Comment by Mno-hime** [comment]
> AC adapter is always "online", even when the cable is disconnected and laptop is running on battery (if I read the attribute correctly):

The problem seems to be that "AC Adapter" information is not being refreshed, see the "Refreshed" attribute.

Energy attributes' values should be in mW & mWh (`lshal` reports that unit), voltage in mV (though I don't see that unit in `lshal`).

Tested again with the new images, and otherwise it looks fine to me.

**Comment by pyhalov** [comment]
Information is refreshed when it's changed. If there's no change, there's nothing to refresh. If device is permanently online,  there are no changes.

**Comment by pyhalov** [comment]
I've updated upower - now it considers devices updated on each poll. Also converted mWh to Wh.

**Pull request merged**